src/config/configure.in

```diff
@@ -437,6 +438,17 @@ AS_HELP_STRING([--enable-device-malloc-async],
 [hypre_using_device_malloc_async=no]
 )
 
+AC_ARG_ENABLE(thrust-nosync,
+AS_HELP_STRING([--enable-thrust-nosync],
+               [Use thrust par_nosync policy (default is NO).]),
+[case "${enableval}" in
+    yes) hypre_using_thrust_nosync=yes ;;
+    no)  hypre_using_thrust_nosync=no ;;
+    *)   hypre_using_thrust_nosync=no ;;
```

Comment:
@liruipeng Would it make sense to enable this by default when cuda >= 12 is detected?

Reply:
I would like to wait a bit. It seems "too new for me", and I am still testing the performance of it. For all, from CUDA 12, there is a new "no sync" execution of thrust (GPU kernels), which is supposed to remove kernel sync. cost. But I am happy to chat more about it.

Reply:
Sounds a good plan to me!